tokio/src/lib.rs

```diff
@@ -401,8 +401,15 @@ compile_error! {
 #[doc(hidden)]
 pub mod macros;
 
-cfg_fs! {
-    pub mod fs;
+#[cfg(feature = "fs")]
+#[cfg_attr(docsrs, doc(cfg(feature = "fs")))]
+pub mod fs;
+
+cfg_fs_internal! {
+    #[cfg(not(feature = "fs"))]
+    #[allow(dead_code)]
+    #[allow(unreachable_pub)]
+    pub(crate) mod fs;
 }
```

Comment:
I don't think this is necessary considering that `tokio::io::{stdin,stdout}` have already been implemented by this strategy without using `tokio::fs`.

Reply:
Thanks for the tip, let me see if I can adapt that approach

Reply:
Worked out beautifully, thanks! (plus it fixed whatever other stack corruption bug I introduced so double thanks!)